autoload/sj/yaml.vim

```diff
@@ -139,5 +139,87 @@ function! s:GetChildren(line_no)
   endwhile
   let next_line_no = next_line_no - 1
 
+  " Preserve trailing empty lines
+  while sj#BlankString(getline(next_line_no)) && next_line_no > line_no
+    let next_line_no = next_line_no - 1
+  endwhile
+
   return [sj#GetLines(line_no + 1, next_line_no), next_line_no]
 endfunction
+
+function! s:splitArrayLines(array)
+  let lines = []
+
+  let line = ''
+  let fences = { '"': '"', "'": "'", '{': '}' }
+
+  for chunk in split(a:array, ',')
+    if line != ''
+      " Inside a string
+      let line = line . ',' . chunk
+
+      if chunk =~ fences[line[0]] . '\s*$'
+        " End of string
+        call add(lines, s:convertToSingleLine(line))
+        let line  = ''
+      endif
+
+    " Start of string
```

Comment:
Maybe "Start of fenced area"? Technically, `{}` is not a string, so that might be a bit misleading.

Reply:
You are right.
(Initially I only handles strings, and added the braces case later)